### DISCUSSION TITLE
Load ToneHunt.org models into ToobAmp ML Amplifier

### DISCUSSION THREAD

**Discussion by danmigdev** [rerdavies/pipedal#86] [Ideas]
Would be possible to load amp models from ToneHunt.org into ToobAmp ML Amplifier plugin? ToneHunt is made for Neural Amp Modeler but I see that the base engine should be the same (RTNeural) so there should be some kind of compatibility already in place, isn't it?

**Comment by 38github** [comment]
Probably not since the NAM models use mostly Wavenet and ToobAmp ML Amplifier seem to use LSTM with a hidden size of only 20.

**Reply by rerdavies** [reply]
Implemented in the new v1.1 release. 

A plugin that does exactly that is available from here:

    https://github.com/mikeoliphant/neural-amp-modeler-lv2

Unfortunately, it had some bugs, that prevent it from working with Pipedal 1.1. I have submitted a pull request that author will accept. I expect it to be merged very shortly.

If you're impatient, the fork which the pull request comes from is this:

    https://github.com/rerdavies/neural-amp-modeler-lv2

but you should use the original once the pull request has been accepted and merged. I don't plan on maintaining my fork any longer than it takes for Mike Oliphant to merge the pull request You'll have to build both of them yourself.

Totally thrilling feature. The models sound amazing! Thanks for letting me know about this. 

CPU use is ferocious. But patches do run stably with no overruns, at about 55% CPU use; but they run! Mike Oliphant (author of the plugin) says that there are some smaller .nam models that run with lower CPU use, but I'm not sure how one tells which are which.

**Reply by 38github** [reply]
> Mike Oliphant (author of the plugin) says that there are some smaller .nam models that run with lower CPU use, but I'm not sure how one tells which are which.

I have several "less cpu" and "even less cpu" versions which you can find at http://coginthemachine.ddns.net/mnt/nam/

**Comment by rerdavies** [comment]
Just to follow up on this thread.... 

PiPedal now includes the TooB Neural Amp Modeller, which allows you to use .NAM files from Tonehunt.org.

(and I also worked with Mike Oliphant t get his port of the Neural Amp Modeler working with PiPedal: https://github.com/mikeoliphant/neural-amp-modeler-lv2 . The two are substantially similar, but TooB Neural Amp Modeller preserves the MIT licensing of the underlying Neural Amp Modeller library that is used by both plugins)
